tests/run-make/backtrace/line-tables-only/test.rs

```diff
@@ -0,0 +1,50 @@
+// ignore-tidy-linelength
+#![feature(backtrace_frames)]
+
+use std::backtrace::{self, Backtrace};
+use std::ffi::c_void;
+use std::ptr::addr_of_mut;
+
+pub type Callback = extern "C" fn(data: *mut c_void);
+
+#[link(name = "callback", kind = "static")]
+extern "C" {
+    pub fn foo(cb: Callback, data: *mut c_void);
+}
+
+extern "C" fn store_backtrace(data: *mut c_void) {
+    let bt = backtrace::Backtrace::capture();
+    unsafe { *data.cast::<Option<Backtrace>>() = Some(bt) };
+}
+
+fn assert_contains(
+    backtrace: &Backtrace,
+    expected_name: &str,
+    expected_file: &str,
+    expected_line: u32,
+) {
+    // FIXME(jieyouxu): fix this ugly fragile test when `BacktraceFrame` has accessors like...
+    // `symbols()`.
+    let backtrace = format!("{:#?}", backtrace);
+    eprintln!("{}", backtrace);
+    assert!(backtrace.contains(expected_name), "backtrace does not contain expected name {}", expected_name);
+    assert!(backtrace.contains(expected_file), "backtrace does not contain expected file {}", expected_file);
+    assert!(backtrace.contains(&expected_line.to_string()), "backtrace does not contain expected line {}", expected_line);
+}
+
+/// Verifies that when debug info includes only lines tables the generated
+/// backtrace is still generated successfully. The test exercises behaviour
+/// that failed previously when compiling with clang -g1.
+///
+/// The test case uses C rather than rust, since at that time when it was
+/// written the debug info generated at level 1 in rustc was essentially
+/// the same as at level 2.
```

Comment:
...Wait, now I'm confused. That's no longer the case, no? Should this test be changed now? :thinking:

Reply:
Good point... Rust *has* `-Cdebuginfo=line-tables-only` now, so IIUC this test can straight up with a UI test with an auxillary crate...